shell/platform/android/io/flutter/embedding/engine/systemchannels/NavigationChannel.java

```diff
@@ -43,6 +45,13 @@ public void pushRoute(@NonNull String route) {
     channel.invokeMethod("pushRoute", route);
   }
 
+  public void pushRouteInformation(@NonNull String route) {
+    Log.v(TAG, "Sending message to push route information '" + route + "'");
+    Map<String, String> message = new HashMap<>();
+    message.put("location", route);
+    channel.invokeMethod("pushRouteInformation", message);
+  }
+
   public void popRoute() {
```

Comment:
This is still used by FlutterView, I am not sure if we can change the API or not.

Reply:
I think this needs to stay for now. I am not following why you want it removed.

Reply:
oh sorry, i highlighted the wrong one, I meant the pushRoute method in line 43

Reply:
Can you replace the usages in flutterView with your new method? If so then maybe the right pattern is to modify the existing one to have the new behavior.

Reply:
yes If i am free to change this API, I can do that. I assume there is no public access to this class from third party plugin then?

Reply:
I dont know how I would know that. I think it would be unusual for us to expose the internals of a plugin but I guess it is possible. @bparrishMines any advice you can share for how to know if this is used by a third party. I checked google3 and did not find any direct references.

Reply:
I will go ahead and merge this pr for now, we can spawn another pr to remove the push route api if we are sure we can remove it

Reply:
I don't know if this is used by a plugin, but it looks like something that could be used used by [Add to App](https://docs.flutter.dev/development/add-to-app).